nwbwidgets/utils/dynamictable.py

```diff
@@ -1,21 +1,41 @@
 from pynwb.core import DynamicTable
+from hdmf.common.table import VectorData
 import numpy as np
+from typing import Iterable
 
 
-def infer_categorical_columns(dynamic_table: DynamicTable):
+def infer_categorical_columns(dynamic_table: DynamicTable, region: Iterable = None):
+    """
+    Parameters
+    ----------
+    dynamic_table: DynamicTable
+    region: Iterable
+        row indices to select
 
+    Returns
+    -------
+    categorical_cols: dict()
+        keys: as columns that are categorical, values as the unique values
+    """
     categorical_cols = {}
+    region = region if region is not None else list(range(len(dynamic_table)))
     for name in dynamic_table.colnames:
         if len(dynamic_table[name].shape) == 1:
-            try:  # TODO: fix this
-                unique_vals = np.unique(dynamic_table[name].data)
-                if 1 < len(unique_vals) <= (len(dynamic_table[name].data) / 2):
+            try:
+                if isinstance(dynamic_table[name].data[0], (str, int, float)):
+                    column_data = [dynamic_table[name].data[i] for i in region]
+                elif hasattr(dynamic_table[name].data[0], 'name'):
```

Comment:
This is particularly useful for example if we want column name 'group' containing `ElectrodeGroup`s to be a category. We can get `ElectrodeGroup.name` to categorize traces (this is a possibility in both `Units `and `Electrodes `table which contain `ElectrodeGroup` in a column)

Reply:
sure, this makes sense